### ISSUE TITLE
[BUG] NameError in ImplicitCF

### ISSUE THREAD

**Issue by miguelgfierro** [recommenders-team/recommenders#2064]
### Description

```
2024-02-19T18:34:57.2553239Z     @pytest.mark.gpu
2024-02-19T18:34:57.2568702Z     def test_model_lightgcn(deeprec_resource_path, deeprec_config_path):
2024-02-19T18:34:57.2569997Z         data_path = os.path.join(deeprec_resource_path, "dkn")
2024-02-19T18:34:57.2571489Z         yaml_file = os.path.join(deeprec_config_path, "lightgcn.yaml")
2024-02-19T18:34:57.2572675Z         user_file = os.path.join(data_path, r"user_embeddings.csv")
2024-02-19T18:34:57.2573837Z         item_file = os.path.join(data_path, r"item_embeddings.csv")
2024-02-19T18:34:57.2574736Z     
2024-02-19T18:34:57.2575330Z         df = movielens.load_pandas_df(size="100k")
2024-02-19T18:34:57.2576298Z         train, test = python_stratified_split(df, ratio=0.75)
2024-02-19T18:34:57.2577150Z     
2024-02-19T18:34:57.2577757Z >       data = ImplicitCF(train=train, test=test)
2024-02-19T18:34:57.2578784Z E       NameError: name 'ImplicitCF' is not defined
2024-02-19T18:34:57.2579411Z 
2024-02-19T18:34:57.2580002Z tests/smoke/recommenders/recommender/test_deeprec_model.py:251: NameError
```

### How do we replicate the issue?

https://github.com/recommenders-team/recommenders/actions/runs/7963399372/job/21738878728

### Other Comments
FYI @SimonYansenZhao this is similar to the issue we got in #2022

**Comment by miguelgfierro** [comment]
`$ pytest  tests/unit/recommenders/models/test_deeprec_model.py::test_lightgcn_component_definition` working with TF 2.8.4 and CUDA 11.6

**Comment by miguelgfierro** [comment]
Still the same error in nightly GPU: https://github.com/recommenders-team/recommenders/actions/runs/8372395934/job/22923293913

**Comment by SimonYansenZhao** [comment]
Resolved in #2077
